### PR TITLE
Feature/vehicle info plugin

### LINKED ISSUE #2189: [SDL 0168] RPC design refactoring
Proposal: [RPC design refactoring](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0168-rpc-design-refactoring.md)

> This proposal is to rework the design of the SDL core application manager layer in order to:
 - Reduce coupling between components
 - Reduce responsibility of the ApplicationManager class
 - Provide a plugin architecture for adding additional functionality and RPC groups easily.

Review: smartdevicelink/sdl_evolution#486

Steering Committee Decision:
> The Steering Committee voted to accept this proposal with the following revision: extract both `Remote control` and `Vehicle data` groups into separate plugins from the other RPCs.

> The proposal .md was updated on May 16, 2018 to include the agreed upon revisions.

### PR DESCRIPTION
**Pull request by olohvinenko**
Fixes #2189

This PR is **ready** for review.

### Risk
This PR makes **major** API changes.

### Testing Plan
TODO: ATF tests and unit tests.

### Summary
Moved related to **VehicleInfo** functionality to separate plugin due to proposal [0168](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0168-rpc-design-refactoring.md).

### Tasks Remaining:
- [ ] [Implement AFT tests]
- [x] [Implement Unit tests]

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/vehicle_info_hmi_command_factory.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  bool IsAbleToProcess(
+      const int32_t function_id,
+      const app_mngr::commands::Command::CommandSource source) const OVERRIDE;
+
+private:
+  app_mngr::ApplicationManager& application_manager_;
+  app_mngr::rpc_service::RPCService& rpc_service_;
+  app_mngr::HMICapabilities& hmi_capabilities_;
+  policy::PolicyHandlerInterface& policy_handler_;
+
```
Comment (KVGrygoriev) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/include/vehicle_info_plugin/vehicle_info_hmi_command_factory.h]: would be nice move variables to parent class

**Comment by LitvinenkoIra** [comment]
looks good to me
waiting for the final version for approval

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "vehicle_info_plugin/vehicle_info_app_extension.h"
+
+CREATE_LOGGERPTR_GLOBAL(logger_, "Vehicle Info Plugin")
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc]: @olohvinenko please avoid spaces in logger names

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+namespace vehicle_info_plugin {
+
+VehicleInfoAppExtension::VehicleInfoAppExtension(application_manager::AppExtensionUID uid)
+  : app_mngr::AppExtension(uid) {
+  LOG4CXX_AUTO_TRACE(logger_);
+}
+
+VehicleInfoAppExtension::~VehicleInfoAppExtension() {}
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc]: @olohvinenko I think autotraces in all this fulctions won't be redundant

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc (hunk 2 above)
Comment (olohvinenko) [reply src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc]: Fixed in 7080abd.

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc (hunk 3 above)
Comment (olohvinenko) [reply src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_app_extension.cc]: Fixed in 7080abd.

**Comment by olohvinenko** [comment]
Rebuild required.

**Comment by vkushnirenko-luxoft** [comment]
Rebuild required

**Pull request merged**